sdk/python/feast/registry.py

```diff
@@ -69,54 +67,62 @@ def __init__(self, registry_path: str, repo_path: Path, cache_ttl: timedelta):
             raise Exception(
                 f"Registry path {registry_path} has unsupported scheme {uri.scheme}. Supported schemes are file and gs."
             )
-        self.cached_registry_proto_ttl = cache_ttl
+        # Read cache from the RegistryStore.
+        self.cached_registry_proto = self._get_registry_proto()
         return
 
     def _initialize_registry(self):
-        """Explicitly forces the initialization of a registry"""
-        self._registry_store.update_registry_proto(updater=None)
+        """Explicitly initializes the registry with an empty proto."""
+        self.cached_registry_proto = RegistryProto()
+        self.cached_registry_proto.registry_schema_version = REGISTRY_SCHEMA_VERSION
+        self._registry_store.update_registry_proto(self.cached_registry_proto)
 
-    def apply_entity(self, entity: Entity, project: str):
+    def apply_entity(self, entity: Entity, project: str, flush: bool = True):
         """
         Registers a single entity with Feast
 
         Args:
             entity: Entity that will be registered
             project: Feast project that this entity belongs to
+            flush: True if the change should be persisted immediately
         """
         entity.is_valid()
         entity_proto = entity.to_proto()
         entity_proto.spec.project = project
+        found_feature_view = False
 
-        def updater(registry_proto: RegistryProto):
-            for idx, existing_entity_proto in enumerate(registry_proto.entities):
-                if (
-                    existing_entity_proto.spec.name == entity_proto.spec.name
-                    and existing_entity_proto.spec.project == project
-                ):
-                    del registry_proto.entities[idx]
-                    registry_proto.entities.append(entity_proto)
-                    return registry_proto
-            registry_proto.entities.append(entity_proto)
-            return registry_proto
+        for idx, existing_entity_proto in enumerate(
+            self.cached_registry_proto.entities
+        ):
+            if (
+                existing_entity_proto.spec.name == entity_proto.spec.name
+                and existing_entity_proto.spec.project == project
+            ):
+                found_feature_view = True
```

Comment:
Sorry what does finding feature views have to do with the entity updating function?

Reply:
Good point. I accidentally named all the booleans the same thing; I'll fix this.